### PR TITLE
fix(mp-3m1c): add missing compId to viewer match aggregators

### PR DESCRIPTION
**Pull request by gitrgoliveira**
## Summary

- Fix viewer match aggregators missing `compId`, which caused `useTeamLineups` to never fetch team lineups for bracket/pool matches in the public viewer
- Fix bout rows displaying team names instead of position numbers when scored via quick-score (frontend filter + backend root cause)
- Fix `c.format` missing from `allMatches` useMemo dependency array (stale `compFormat` on SSE-driven format changes)
- Fix bracket-tab `onMatchClick` omitting `roundIndex`, which forced `useTeamLineups` to parse the round label string instead of using the 0-based index
- **Backend root cause:** quick-score handler copied team names into every synthesised sub-bout's `SideA`/`SideB` — these fields should be empty since individual bout sides are unknown in quick-score mode. Also harden the `sub.Winner == sub.SideA` fallback in `computeStandings`, `ComputeTeamSummary`, and `deriveDaihyosenWinner` against the `"" == ""` false-positive.
- **UX polish (critique P2/P3):** Focus trap + `aria-modal` on MatchViewerModal; Escape-to-close on inline MatchDetailCard; competition name in card header; waza tooltips on ippon slots; `<abbr>` with title on IV/PW summary labels; bout labels prefixed with `#` for clarity; `msb-lab` font bump 8px to 10px; TV completed-row tint; TV bout winner slot emphasis.

## Why

**compId bug:** `useTeamLineups` derives `compId` from `competition?.id || match?.compId`. `MatchDetailCard` passes `undefined` for `competition`, so it relies entirely on `match.compId`. Three match enrichment sites in `ViewerCompetition` omitted `compId`, causing `useTeamLineups` to early-return without fetching.

**Bout-row bug (frontend):** `subSideName` in `BoutSubRow` returned the team name from `sub.sideA`/`sub.sideB` (stored by the quick-score backend path), so the `boutNum` fallback never triggered. Fix: filter out match-level team names so the fallback chain reaches the position number.

**Bout-row bug (backend root cause):** The quick-score handler (`PUT /competitions/:id/matches/:mid/quick-score`) set `SideA: req.SideA, SideB: req.SideB` on every synthesised `SubMatchResult`, propagating team names where individual competitor names or empty strings belong. The standings code (`computeStandings`, `ComputeTeamSummary`) had a `sub.Winner == sub.SideA` fallback that only worked correctly when sub.SideA was non-empty — an empty-Winner draw with empty sub.SideA would false-positive as a sideA win via `"" == ""`. Fixed by: (1) leaving sub-bout SideA/SideB empty in quick-score, (2) guarding all 3 fallback sites with `sub.SideA != ""`.

**useMemo dep:** `c.format` was used in pool enrichment (`compFormat: c.format`) but omitted from the dependency array — if competition format changes via SSE without other deps changing, `compFormat` would be stale.

**roundIndex:** bracket-tab `onMatchClick` built `selectedMatch` without `roundIndex`, so `useTeamLineups` fell back to parsing `match.round` (now a bracket-size label like "Round 16" that a "Round N"→N-1 parse would misread as round 15).

**UX polish:** Nielsen heuristic critique (27/40) identified P2 accessibility gaps (no focus trap, no keyboard dismiss) and P3 discoverability issues (cryptic IV/PW labels, no competition context, low contrast on completed rows). All addressed.

## Files changed

| File | What |
|---|---|
| `web-mobile/js/viewer.jsx` | Add `compId: c.id` to pool/bracket aggregators + `onMatchClick`; add `c.format` to useMemo deps; add `roundIndex: ri` to bracket click handler; focus trap + aria-modal on MatchViewerModal; Escape-to-close + comp name on MatchDetailCard |
| `web-mobile/js/match_scoreboard.jsx` | Add `matchSideA`/`matchSideB` props to `BoutSubRow`; filter team names from sub-bout side names to fall through to position numbers; `#`-prefixed bout labels; waza tooltips on ippon slots; `<abbr>` with title on IV/PW summary labels |
| `web-mobile/js/display.jsx` | Thread `matchSideA`/`matchSideB` to TV display `TeamScoreboard` call; completed-row tint on TvIndividualBoard |
| `web-mobile/css/styles.css` | `.match-detail-card__comp` styling; `msb-lab` font bump 8px to 10px; `abbr.msb-lab` decoration reset; TV bout winner slot emphasis (3.2vh) |
| `web-mobile/js/__tests__/match_scoreboard.test.jsx` | Add 2 tests: team-name filtering falls through to bout number; real competitor names still render; updated bout label assertions for `#` prefix |
| `internal/mobileapp/handlers_match.go` | Quick-score: stop copying team names into sub-bout SideA/SideB |
| `internal/mobileapp/handlers_match_test.go` | Assert quick-score sub-bouts have empty SideA/SideB |
| `internal/engine/scoring.go` | Guard `sub.Winner == sub.SideA` with `sub.SideA != ""` in computeStandings + deriveDaihyosenWinner |
| `internal/engine/daihyosen.go` | Guard `sub.Winner == sub.SideA` with `sub.SideA != ""` in ComputeTeamSummary |
| `internal/engine/scoring_test.go` | New test: empty sub-bout sides with empty Winner count as draws, not false sideA wins |
| `internal/engine/pool_daihyosen_test.go` | Update comment to reflect new guard |

## Screenshots

Viewer MatchDetailCard for a completed team match (Team Beta 2-1 Team Alpha) at mobile viewport (390x844). Bout rows show position numbers (#1, #2, #3), ippon letters (K, M, H) render correctly, IV/PW summary header present.

Viewer MatchDetailCard with team match detail

## Test plan

- [x] `make go/test` passes (lint + security scan + tests) — all packages above 85% coverage gate
- [x] New/updated unit tests cover the change — 2 vitest cases + 2 Go tests (quick-score empty sides, empty-sides-draw-not-false-win)
- [x] `npx vitest run` — 60 test files, 1441 tests pass (includes updated bout label `#` prefix assertions)
- [x] Manual browser verification (for `web-mobile/` changes) — created team tournament (4 teams, 3-person, playoffs), scored match m-r1-0 (Team Beta 2-1 Team Alpha) via detailed score API, clicked completed match in viewer → MatchDetailCard rendered with bout position numbers (not team names), ippon letters (K, M, H), and IV/PW summary
- [x] Screenshots added above (REQUIRED for any UI-affecting change)
- [x] No new console errors or warnings

Closes mp-3m1c

### CODE REVIEW

File: internal/mobileapp/handlers_match.go [hunk 1]
```diff
@@ -335,6 +335,15 @@ func RegisterMatchHandlers(r *gin.RouterGroup, eng *engine.Engine, store Competi
...
 		}
+		if req.TeamAWins < 0 || req.TeamBWins < 0 || req.Draws < 0 {
+			c.JSON(http.StatusBadRequest, gin.H{"error": "win/draw counts must be non-negative"})
+			return
+		}
+		const maxBouts = 100
+		if req.TeamAWins+req.TeamBWins+req.Draws > maxBouts || req.TeamAWins+req.TeamBWins+req.Draws < 0 {
+			c.JSON(http.StatusBadRequest, gin.H{"error": "total bout count exceeds maximum"})
+			return
+		}
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply internal/mobileapp/handlers_match.go]: Fixed: each count is now individually validated against `maxBouts` before any addition, so even on 32-bit platforms the sum cannot overflow. The validated total is computed once and checked separately.

Class swept — the `make([]SubMatchResult, 0, total)` at handlers_match.go:367 is the only consumer and is now safe (each field ≤ 100, sum ≤ 300). No other uncapped-int-sum patterns in this PR diff.

File: internal/mobileapp/handlers_match.go [hunk 2]
```diff
@@ -335,6 +335,20 @@ func RegisterMatchHandlers(r *gin.RouterGroup, eng *engine.Engine, store Competi
...
+		if req.TeamAWins > maxBouts || req.TeamBWins > maxBouts || req.Draws > maxBouts {
+			c.JSON(http.StatusBadRequest, gin.H{"error": "total bout count exceeds maximum"})
+			return
+		}
+		total := req.TeamAWins + req.TeamBWins + req.Draws
+		if total > maxBouts {
+			c.JSON(http.StatusBadRequest, gin.H{"error": "total bout count exceeds maximum"})
+			return
+		}
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply internal/mobileapp/handlers_match.go:351]: Fixed: per-field check now says "individual bout count exceeds maximum"; the sum check retains "total bout count exceeds maximum". Class swept — no other misleading error messages in the validation block.

File: internal/mobileapp/handlers_match.go [hunk 3]
```diff
@@ -352,20 +366,23 @@ func RegisterMatchHandlers(r *gin.RouterGroup, eng *engine.Engine, store Competi
...
 
 		// Synthesise SubResults so standings IV/IL/IT counts are correct.
-		// SideA/SideB must be set so the empty-Winner draw case doesn't
-		// accidentally match `sub.Winner == sub.SideA` in computeStandings.
+		// Sub-bout SideA/SideB are left empty — individual bout sides are
+		// unknown in quick-score mode (no lineup). Winner attribution in
+		// computeStandings uses `sub.Winner == m.SideA` (the match-level
+		// name); the `sub.Winner == sub.SideA` fallback is guarded against
+		// the "" == "" false-positive.
 		subResults := make([]state.SubMatchResult, 0, req.TeamAWins+req.TeamBWins+req.Draws)
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply internal/mobileapp/handlers_match.go]: Fixed: `make([]state.SubMatchResult, 0, total)` now uses the already-validated `total` variable instead of recomputing the sum. Class swept — this was the only raw-sum consumer remaining.

File: internal/engine/scoring_test.go [hunk 4]
```diff
@@ -215,6 +215,52 @@ func TestScoreSummary_Team(t *testing.T) {
...
+
+	teamA := pool[0]
+	assert.Equal(t, 1, teamA.IndividualWins, "sub with Winner=TeamA → IV for A")
+	assert.Equal(t, 1, teamA.IndividualLosses, "sub with Winner=TeamB → IL for A")
+	assert.Equal(t, 1, teamA.IndividualDraws, "sub with empty Winner+empty SideA → draw, not false win")
+
+	teamB := pool[1]
+	assert.Equal(t, 1, teamB.IndividualWins)
+	assert.Equal(t, 1, teamB.IndividualLosses)
+	assert.Equal(t, 1, teamB.IndividualDraws, "sub with empty Winner+empty SideB → draw, not false win")
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply internal/engine/scoring_test.go]: Fixed: both teams have identical stats (1 IV, 1 IL, 1 IT) so sort order is non-deterministic. Changed to look up entries by `Player.Name` instead of index. Class swept (`pool[0]`/`pool[1]` pattern in PR diff) — no other occurrences in this diff.

File: web-mobile/js/match_scoreboard.jsx [hunk 5]
```diff
@@ -180,9 +186,17 @@ function centreMarks(sub) {
...
+  const subSideName = (v) => {
+    const n = (v && v.name) || (typeof v === "string" ? v : "");
+    if (!n) return "";
+    // Filter out match-level team names — when the backend stores the team
+    // name in every sub-bout (quick-score path), we must fall through to the
+    // bout number rather than repeating the team name on every row.
+    if (n === matchSideA || n === matchSideB) return "";
+    return n;
+  };
+  const boutNum = isDH ? "DH" : "#" + (sub && sub.position > 0 ? sub.position : index + 1);
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/match_scoreboard.jsx:196]: Fixed in 1ea5a7f: added `matchSideA`/`matchSideB` parameters to `centreMarks` as a third fallback in the winner comparison chain (`sub.sideA` → `sub.teamA` → `matchSideA`). `BoutSubRow` now passes them through. Class swept — the other `centreMarks` call site (`IndividualScore`) synthesizes `sub` with match-level sides already populated, so no sibling.

File: web-mobile/js/display.jsx [hunk 6]
```diff
@@ -642,9 +645,11 @@ function TvDisplay({ court, tournament, competitions, withZekkenName, connected
         // teamIVPW already prefers an explicit `sub.winner` (which the server
         // guarantees equals sideA/sideB), so a hantei-decided 0-0 bout is
         // counted as an IV for its winner there — no extra hantei loop needed.
-        const { ivShiro, ivAka, pwShiro, pwAka } = teamIVPW(subResults);
+        const mA = promoted.match.sideA?.name || (typeof promoted.match.sideA === "string" ? promoted.match.sideA : "");
+        const mB = promoted.match.sideB?.name || (typeof promoted.match.sideB === "string" ? promoted.match.sideB : "");
+        const { ivShiro, ivAka, pwShiro, pwAka } = teamIVPW(subResults, mA, mB);
         return ivShiro === ivAka && pwShiro === pwAka;
-    }, [subResults, isTeamMatch, isKnockoutPhase]);
+    }, [subResults, isTeamMatch, isKnockoutPhase, promoted]);
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/display.jsx]: Fixed: extracted `promotedSideA`/`promotedSideB` as stable string primitives before the memo, then depended on those instead of the `promoted` object literal. Class swept — no other `useMemo` in this diff deps on `promoted` directly.

File: web-mobile/js/viewer.jsx [hunk 7]
```diff
@@ -1832,6 +1832,7 @@ function ViewerCompetition({ tournament, competition, pools, poolMatches, standi
 
 // Inline match detail card — shown directly on the page (no modal needed).
 function MatchDetailCard({ match, onClose }) {
+  window.useEscapeToClose(onClose);
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/viewer.jsx]: Fixed: added `escapeToClose` prop (default true) to `MatchDetailCard`; pass `escapeToClose={false}` from `MatchViewerModal` so the card calls `useEscapeToClose(undefined)` in that context — modal remains the sole Escape handler. Sibling sweep: other inline card usages (lines 2096/2111/2130) are standalone with no parent Escape handler — correct to leave them wired.
